execute.go

```diff
@@ -70,15 +70,29 @@ func executeWithOutputChan(outputChan chan string, env []string, workingDir stri
 		return nil, fmt.Errorf("error starting process: %v", startErr)
 	}
 
-	waitErr := cmd.Wait()
-
-	if waitErr != nil {
-		if exitErr, ok := waitErr.(*exec.ExitError); ok {
-			if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
-				exitCode = status.ExitStatus()
+	//create a new effectiveconfig instance
+	var con = new(effectiveConfig)
+
+	// 1 deep channel for done
+	done := make(chan error, 1)
+	go func() {
+		done <- cmd.Wait()
+	}()
+	select {
+	case <-time.After(con.BuildTimeout()):
+		if err := cmd.Process.Kill(); err != nil {
+			return nil, fmt.Errorf("Failed to kill process: %v", err)
+		}
+		<-done
+	case err := <-done:
+		if err != nil {
+			if exitErr, ok := cmd.Wait().(*exec.ExitError); ok {
+				if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+					exitCode = status.ExitStatus()
```

Comment:
This is getting a little too complex.  It might be worthwhile to pull the `select` out into its own function and/or the `case`s.